app/src/main/java/com/alexbar/layoutbasic/movies_app/networking/RetrofitInstance.kt

```diff
@@ -0,0 +1,33 @@
+package com.alexbar.layoutbasic.movies_app.networking
+
+import com.alexbar.layoutbasic.movies_app.utils.MovieConstants.api_base_url
+import com.alexbar.layoutbasic.movies_app.utils.MovieConstants.user_token
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object RetrofitInstance {
+    private val logging = HttpLoggingInterceptor().apply {
+        level = HttpLoggingInterceptor.Level.BODY
+    }
+    private val tokenInterceptor = Interceptor { chain ->
+        val newRequest = chain.request().newBuilder()
+            .addHeader("Authorization", "Bearer $user_token")
+            .build()
+        chain.proceed(newRequest)
+    }
```

Comment:
@ezatpanah For these specific endpoints, the API requires a user token, and I found that I could implement it in this way. However, let's suppose that for some reason there are some endpoints where I shouldn't send the token. How could I achieve that? As I understand it, by implementing it this way, all requests would be sent with this token, correct?

Reply:
With your implementation, all requests made using the RetrofitInstance will include the user token because the token interceptor is added to the OkHttpClient used by Retrofit. but regarding your question , ypu can use this approach :
```

private val tokenInterceptor = Interceptor { chain ->
    val originalRequest = chain.request()
    val request = if (isTokenRequired(originalRequest.url().toString())) {
        originalRequest.newBuilder()
            .addHeader("Authorization", "Bearer $user_token")
            .build()
    } else {
        originalRequest
    }
    chain.proceed(request)
}

private fun isTokenRequired(url: String): Boolean {
    return !url.contains("specific_endpoint_not_requiring_token")
}

```

 the interceptor checks if the URL requires a token. You can define the logic in the isTokenRequired function to determine whether a token is required for a given URL. If the URL requires a token, the interceptor adds the Authorization header with the token. Otherwise, it proceeds without adding the header.